packages/docs/content/docs/prompting-guideline.mdx

```diff
@@ -0,0 +1,149 @@
+---
+title: Prompting Guideline
+description: A guide to effective prompt engineering with Pochi
+icon: "Lightbulb"
+---
+
+# Pochi Prompting Guide
+
+## Intro
+
+Welcome to Pochi Prompting Guideline! Please consider using this doc as reference to help you write more effective prompts, guiding Pochi to better understand your instructions, and thus maximizing your dev experience with Pochi.
+
+## Overall Prompting Guidelines
+
+- **Provide Clear Context:** Use `@` to reference files/folders (e.g., `@api.js`) or `README.pochi.md` for project rules and goals (e.g., coding standards, security priorities).
+- **Break Down Tasks:** Split complex problems into steps for stability (e.g., review, then optimize).
+- **Ensure Consistency:** Rely on structured prompts (XML/JSON) for repeatable, predictable outputs.
+- **Iterate Safely:** Tweak prompts step-by-step. Using structured formats like XML or JSON for your prompts makes it easier to track and revert changes if needed.
+- **Leverage Pochi Features:**
+  - Use `~/.pochi/config.jsonc` for [models](/models) & [MCPs](/mcp).
+  - Use `README.pochi.md` for [project-specific rules](/rules).
+  - Use `.pochi/workflows/` for [custom workflows](/workflows).
+  - Use `.pochi/agents/` for [custom agents](/custom-agent).
+  - Use [Pochi's sharing features](/share) to share task histories.
+  - Use the [GitHub](/github) & [Slack](/slack) integrations for team collaboration.
+
+## How to Prompt Pochi
+
+Effective prompting transforms vague ideas into precise actions by emphasizing clarity, examples, and structure. Approach it as directive a collaborative agent that benefits from guided, step-by-step instructions to deliver reliable results.
+
+### Craft Clear Prompts
+
+Be specific and direct. Avoid indirect phrasing like “can you", state the action explicitly.
+
+**Example:** “Fix `@utils.js` line 42: Add input validation for `process(data)` where `data` can be null, return a diff.”
+
+### Use Few Shot Prompts
+
+Provide 2-3 input-output examples to set expectations
+
+**Example:**
+“Input: Error in `@hooks.js` - ‘useEffect is not defined’ in `render()`;
+Output: `import { useEffect } from 'react'; useEffect(() => { console.log('mounted'); }, []);`.”
+
+**Example:**
+ “Input: Slow `@api.js` - 5s load on `/users` due to DB query;
+Output: `const cache = require('node-cache'); cache.set('users', await db.query('SELECT * FROM users'), 3600);`.”
+
+### Let Pochi Think (Chain-of-Thought)
+
+For complex logic, request step-by-step reasoning.
+
+**Example:**
+“Debug `@main.py`: Identify the null reference issue in `fetchUser()`, propose a fix, test with `pytest test_main.py`, log the result."
+
+### Use Structured Prompts
```

Comment:
this is more of a prompting skills for AI app developer (like us), not for our end user.